server/socket/index.ts

```diff
@@ -1,147 +1,104 @@
-
-  wss.on(
-    WebsocketMessageType.Notification,
-    (payload: IWebsocketsPayload<any>, notifications: { [user: number]: any }) => {
-      if (logging) {
-        log.info(`Payloading ${JSON.stringify(payload)} to users ${JSON.stringify(Object.keys(notifications))}`);
-      }
-      for (const [ user, notification ] of Object.entries(notifications)) {
-        if (user && user in userMap && userMap[user].isAlive) {
-          // augment notification with unique database id
-          payload.data.id = notification.id;
-          payload.data.subscription_id = notification.subscription_id;
-          userMap[user].send(JSON.stringify(payload));
-        }
-      }
-    }
+export function setupWebSocketServer(httpServer: http.Server) {
+	// since the websocket servers are not linked with the main Commonwealth server we do not send the socket.io client
+// library to the user since we already import it + disable http long-polling to avoid sticky session issues
+	const io = new Server(httpServer, { transports: ['websocket'], cors: {
+			origin,
+			methods: ["GET", "POST"]
+		}});
+
+	io.use(authenticate);
+
+	io.on('connection', (socket) => {
+		log.trace(`${socket.id} connected`)
+		socket.on('disconnect', () => {
+			log.trace(`${socket.id} disconnected`);
+		});
+	});
+
+	io.engine.on('connection_error', (err) => {
+		// log.error(err.req);      // the request object
+		// console.log(err.code);     // the error code, for example 1
+		// console.log(err.message);  // the error message, for example "Session ID unknown"
+		// console.log(err.context);  // some additional error context
+		log.error("A WebSocket connection error has occurred", err)
+	})
+
+// create the chain-events namespace
+	const ceNamespace = createCeNamespace(io);
+
+	// enables the admin analytics dashboard (creates /admin namespace)
+	instrument(io, {
+		auth: false,
+	})
+
+	const pool = new Pool({
+		connectionString: DATABASE_URI,
+		ssl: {
+			rejectUnauthorized: false,
+		},
+	})
+	pool.query(`
+  CREATE TABLE IF NOT EXISTS socket_io_attachments (
+      id          bigserial UNIQUE,
+      created_at  timestamptz DEFAULT NOW(),
+      payload     bytea
   );
-
-  wss.on(WebsocketMessageType.ChainEntity, (payload: IWebsocketsPayload<any>) => {
-    if (logging) log.info(`Payloading ${JSON.stringify(payload)}`);
-    for (const [ session, sessionSocket ] of Object.entries(sessionMap)) {
-      if (sessionSocket && sessionSocket.isAlive) {
-        sessionSocket.send(JSON.stringify(payload), (err?) => {
-          if (err) {
-            log.error(`Failed to send chain entity to session: ${session}`);
-            log.error(`Error: ${err}.`);
-            // TODO: remove from map if err is that it's closed?
-          }
-        });
-      }
-    }
-  });
+`).then((res) => {
+		log.info('Socket.io query successful')
+	}).catch((e) => {
+		log.error("Postgres Adapter will not work so cross server websocket rooms will not be available.", e)
+	})
+
+	// @ts-ignore
```

Comment:
What's the ts problem here?

Reply:
`Type 'PostgresAdapter' is not assignable to type 'Adapter'.` Took this code straight from the docs so the code is correct just some type issues in Socket.io library.